composer/loggers/progress_bar_logger.py

```diff
@@ -123,47 +141,65 @@ def should_log(state: State, log_level: LogLevel, console_log_level: LogLevel =
             self.should_log = should_log
         else:
             self.should_log = console_log_level
+
+        # set the stream
         if isinstance(stream, str):
-            if stream.lower() == 'stdout':
-                stream = sys.stdout
-            elif stream.lower() == 'stderr':
-                stream = sys.stderr
-            else:
-                raise ValueError(
-                    f"Invalid stream option: Should be 'stdout', 'stderr', or a TextIO-like object; got {stream}")
+            stream = {
+                'stdout': sys.stdout,
+                'stderr': sys.stderr,
+            }[stream.lower()]
         self.stream = stream
 
-    def log_data(self, state: State, log_level: LogLevel, data: Dict[str, Any]) -> None:
-        if dist.get_local_rank() == 0 and self.is_train in self.pbars:
-            # Logging outside an epoch
-            assert self.is_train is not None
-            self.pbars[self.is_train].log_data(data)
+    @property
+    def current_pbar(self) -> Optional[_ProgressBar]:
+        if self.is_train is None:
+            return None
+        return self.train_pbar if self.is_train else self.eval_pbar
 
-        if not self.should_log(state, log_level):
-            return
-        data_str = format_log_data_value(data)
-        log_str = f'[{log_level.name}][batch={int(state.timestamp.batch)}]: {data_str}'
+    @current_pbar.setter
+    def current_pbar(self, pbar: Optional[_ProgressBar]):
+        if self.is_train is None:
+            raise AssertionError('Cannot set pbar if self.is_train is not set.')
```

Comment:
Since this is a public setter, I think it should be a value error.

```suggestion
            raise RuntimeError('Cannot set pbar if self.is_train is not set.')
```

That said, do we need a public getter and setter for this?